src/components/ChatEntry.js

```diff
@@ -1,22 +1,47 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.js';
 
 const ChatEntry = (props) => {
+  const onLikeButtonClick = () => {
+    const updatedMessage = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked,
+    };
+
+    props.onUpdateMessage(updatedMessage.id);
+  };
+
+  // this is a ternary
+  const heartColor = props.liked ? '❤️' : '🤍';
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp}></TimeStamp>
+        </p>
+        <button className="like" onClick={onLikeButtonClick}>
+          {heartColor}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number,
+  sender: PropTypes.string,
+  body: PropTypes.string,
+  timestamp: PropTypes.string,
+  liked: PropTypes.bool,
+  onUpdateMessage: PropTypes.func,
 };
```

Comment:
Great job with these propTypes. We could potentially require them all, but that does cause some weirdness with the way we set up our tests, so that's on us!